core/src/main/scala/zio/config/ConfigDocs.scala

```diff
@@ -15,8 +15,9 @@ object ConfigDocs {
   final case class Sources(list: List[String])
 
   final case object Empty                                                       extends ConfigDocs[Nothing, Nothing]
-  final case class Path[K, V](path: K, details: Details[V])                     extends ConfigDocs[K, V]
   final case class NestedPath[K, V](path: K, docs: ConfigDocs[K, V])            extends ConfigDocs[K, V]
   final case class Both[K, V](left: ConfigDocs[K, V], right: ConfigDocs[K, V])  extends ConfigDocs[K, V]
   final case class OneOf[K, V](left: ConfigDocs[K, V], right: ConfigDocs[K, V]) extends ConfigDocs[K, V]
+  final case class Sequence[K, V](element: ConfigDocs[K, V])                    extends ConfigDocs[K, V]
+  final case class SequenceWithValue[K, V](element: List[ConfigDocs[K, V]])     extends ConfigDocs[K, V]
```

Comment:
case class Details(v: Option[V], sources: Sources, descriptions; List[String]) extends ConfigDocs[K, V]
.....
case class Sequence(element: List[ConfigDocs[K, V]]) extends ConfigDocs[K, V]

Reply:
Pretty printing this is going to be a bit tricky/non-trivial. 
Not for this PR, but worth mentioning about it, before we hit 1.0

Reply:
done